From 070feb7c5ebd0f2ca721ca5d75bdd3fd8cffe961 Mon Sep 17 00:00:00 2001
From: Baruch Siach <baruch@tkos.co.il>
Date: Fri, 27 Apr 2018 07:34:57 +0300
Subject: [PATCH] proc/sig: fix build for sparc

The code undefines SIGLOST which breaks references to SIGPWR.

Taken from a patch suggested in upstream bug report #93.

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---
Upstream status: https://gitlab.com/procps-ng/procps/issues/93
---
 proc/sig.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/proc/sig.c b/proc/sig.c
index b883185fc28a..6ca9512cc70c 100644
--- a/proc/sig.c
+++ b/proc/sig.c
@@ -52,10 +52,6 @@
 #  undef SIGSTKFLT
 #endif
 
-#if !defined(__GNU__) && defined(SIGLOST)
-#  undef SIGLOST
-#endif
-
 #ifndef SIGRTMIN
 #  warning Standards require that <signal.h> define SIGRTMIN; assuming 32
 #  define SIGRTMIN 32
@@ -87,7 +83,7 @@ static const mapstruct sigtable[] = {
   {"ILL",    SIGILL},
   {"INT",    SIGINT},
   {"KILL",   SIGKILL},
-#ifdef SIGLOST
+#if defined(__GNU__)
   {"LOST",   SIGLOST},  /* Hurd-specific */
 #endif
   {"PIPE",   SIGPIPE},
-- 
2.17.0

